binding/Binding.Shared/PlatformConfiguration.cs

```diff
@@ -49,6 +53,36 @@ static PlatformConfiguration ()
 #endif
 
 			Is64Bit = IntPtr.Size == 8;
+
+			isMuslLazy = new Lazy<bool>(IsMuslImplementation);
 		}
+
+		private static bool IsMuslImplementation()
+		{
+			try
+			{
+				var cpu = RuntimeInformation.ProcessArchitecture;
+				switch (cpu)
+				{
+					case Architecture.X86:
+						return AccessCheck("/lib/libc.musl-x86.so.1", 0) == 0;
```

Comment:
Checking for `/lib/libc.musl-*` seems to be a decent way to detect Musl systems. If all else fails, then we fallback to the existing logic.

Reply:
The existence of that file does **not** indicate the libc variant that's used by the current process.

Reply:
😢 Is there any way to check for Musl? Is there some specific thing that exists in Musl or glibc that we can use to detect? Maybe there is some thing we call that returns a string that says musl or not?

Reply:
What if I told you that technically one system can have _both_?

Reply:
I guess you could try to DllImport `gnu_get_libc_version` from `libc` and see if it throws 
`EntryPointNotFoundException`. If it doesn't, you are definitely running with glibc.

Reply:
BTW, since that check happens at runtime, I'd suggest to expose an API for selecting the proper binary and SkiaSharp-specific environment variable. So detection logic would look like this:

1) use API-provided-override if configured
2) use the value from environment variable if exists
3) attempt to auto-detect

Reply:
Good point. Let me add that.

Reply:
I only added 1. as I don't want a library to be reaching out into the environment. This can of course be added by a consuming library or by the final app. However, the API that I am making public is the existing `PlatformConfiguration` class. This was previously used just to detect the OS, arch or pointer size, however, I added both a `IsMusl` and `IsMuslOverride` which can be used to control if the process is glibc or musl:

```csharp
// use default fallback
PlatformConfiguration.IsMuslOverride = null;
// is Musl
PlatformConfiguration.IsMuslOverride = true;
// is NOT Musl
PlatformConfiguration.IsMuslOverride = false;
```

If there is ever another libc, we can just add a new `IsXlibc` and a `IsXlibcOverride`. Then it still up to the consumer to set the correct values. The recommendation is to only set when you are sure it is one or the other. Although not very high chance we will get another libc, setting it to false when unsure will make the detection checks stop early. 

If SkiaSharp is detecting Musl incorrectly, then it should be seen as a bug and hopefully will get reported and fixed so the override should remain mostly unused.

Reply:
Actually, I am making the API internal for now and can be accessed via reflection. I have tests to make sure it stays.

The reason for this is that this lookup system only exists on mono and we are really starting to move away from that. I added the API for the existing apps that need to control the lookup, but with .NET Core and .NET 5+, all the lookup is done via the runtime features.

And if there is some pluggable system that needs multiple versions of libSkiaSharp in memory, then this is still not an issue because since .NET Core 3.0 there is the `NativeLibrary` which is the preferred and more correct way to override the loading mechanism: https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.nativelibrary

Reply:
Well, it should actually be using the `AssemblyLoadContext` and works with the results of a `Publish` for a project. I created a sample with SkiaSharp based on the docs: https://github.com/mattleibow/SkiaSharpPluggable